apps/common/decorators.py

```diff
@@ -382,3 +384,22 @@ def handle(cache):
 
     return bulk_handle(handle, batch_size, timeout)
 
+
+def check_mfa(func):
+    @wraps(func)
+    def wrapper(self, code):
+        cache_key = self.cache_key(self.name, self.user)
+        cache_code = cache.get(cache_key)
+
+        if cache_code == code:
+            return False, _('Invalid security code')
+
+        ok, msg = func(self, code)
+        if not ok:
+            return False, msg
+
+        cache.set(cache_key, code, 60 * 5)
+        return True, msg
+
+    return wrapper
+
```

Comment:
Since this is a sample code snippet without any specific context, it's difficult to identify any issues or suggest improvements based on current knowledge cutoff. It seems like some functions might be using Django's `transaction` and `cache`, which are not relevant for today.

If you need specific checks related to these topics (like checking transactions being correctly handled), please provide more details so I can accurately assess the potential issues.